asio/CMakeLists.txt

```diff
@@ -0,0 +1,240 @@
+option(ASIO_ENABLE_HANDLER_TRACKING "enable handler tracking" OFF)
+option(ASIO_DISABLE_KQUEUE "disable kqueue" OFF)
+option(ASIO_DISABLE_EPOLL "disable epoll" OFF)
+option(ASIO_USE_CLANG_TIDY "use clang-tidy if found" OFF)
+
+
+#
+# Where to put all the RUNTIME targets when built.  This variable is used
+# to initialize the RUNTIME_OUTPUT_DIRECTORY property on all the targets.
+#
+set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${PROJECT_BINARY_DIR}/bin)
+set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${PROJECT_BINARY_DIR}/lib)
+set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${PROJECT_BINARY_DIR}/lib)
+
+
+set(CPPdefinitions)
+set(EXTRA_LIBS)
+
+if(NOT CONAN_COMPILER AND WIN32)
+  list(APPEND CPPdefinitions _WIN32_WINNT=0x0501 __USE_W32_SOCKETS
+    ASIO_HAS_VARIADIC_TEMPLATES ASIO_DISABLE_CONSTEXPR
+  )
+  set(EXTRA_LIBS wsock32 ws2_32)
+else()
+  set(EXTRA_LIBS ${CONAN_TARGETS})
+endif()
+
+if(NOT CONAN_COMPILER AND ${CMAKE_HOST_SYSTEM_NAME} MATCHES "MINGW.*")
+  list(APPEND CPPdefinitions _WIN32_WINNT=0x0501 __USE_W32_SOCKETS)
+endif()
+
+
+if(ASIO_NO_DEPRECATED)
+  message(STATUS "disable deprected interfaces")
+  list(APPEND CPPdefinitions ASIO_NO_DEPRECATED)
+endif()
+
+if(ASIO_ENABLE_HANDLER_TRACKING)
+  message(STATUS "enable handler tracking")
+  list(APPEND CPPdefinitions ASIO_ENABLE_HANDLER_TRACKING)
+endif()
+
+if(ASIO_DISABLE_KQUEUE)
+  message(STATUS "disble queue")
+  list(APPEND CPPdefinitions ASIO_DISABLE_KQUEUE)
+endif()
+
+if(ASIO_DISABLE_EPOLL)
+  message(STATUS "disable epoll")
+  list(APPEND CPPdefinitions ASIO_DISABLE_EPOLL)
+endif()
+
+if(ASIO_USE_CLANG_TIDY)
+  message(STATUS "use clang-tidy")
+  find_program(CMAKE_CXX_CLANG_TIDY
+    NAMES clang-tidy clang-tidy-8 clang-tidy-9 clang-tidy-7
+    HINTS /usr/local/bin /opt/local/bin
+  )
+else()
+  unset(CMAKE_CXX_CLANG_TIDY CACHE)
+endif()
+
+if(NOT CONAN_COMPILER)
+  if(NOT ASIO_STANDALONE)
+    #####################################################################
+    # use boost if found
+    option(Boost_USE_MULTITHREADED "Set to OFF to use the non-multithreaded" ON)
+    option(Boost_DEBUG "Set to ON to enable debug output from FindBoost." OFF)
+    option(Boost_DETAILED_FAILURE_MSG "Set to ON to get detailed information" ON)
+    # Set Boost_NO_BOOST_CMAKE to ON to disable the search for boost-cmake.
+    set(Boost_NO_BOOST_CMAKE ON)
+    set(BOOST_ROOT "/opt/local" CACHE PATH "path to the boost lib")
+    set(Boost_USE_STATIC_LIBS OFF)
+    if(WIN32)
+      set(Boost_USE_STATIC_RUNTIME OFF)
+    endif()
+    find_package(Boost 1.67
+        COMPONENTS atomic chrono coroutine context thread system
+    )
+    #####################################################################
+
+    if(Boost_FOUND)
+      include_directories(${Boost_INCLUDE_DIRS})
+      link_directories(${Boost_LIBRARY_DIRS})
+      list(APPEND CPPdefinitions BOOST_ALL_NO_LIB=1
+        BOOST_CHRONO_DONT_PROVIDE_HYBRID_ERROR_HANDLING
+      )
+      list(APPEND EXTRA_LIBS ${Boost_LIBRARIES})
```

Comment:
`include_directories` and `link_directories` should not be used, prefer the `target_*` variants of the commands.

Drop the variables and use the `Boost::*` targets instead.

Reply:
Than we need Boost v1.71 at least, not available on debian stable!